From garzik@havoc.gtf.org Fri Dec  2 19:24:13 2005
Date: Fri, 2 Dec 2005 20:39:41 -0500
From: Jeff Garzik <jgarzik@pobox.com>
Cc: <gregkh@suse.de>, <ak@suse.de>
Subject: [PATCH 1/3] x86 PCI domain support: a humble fix
Message-ID: <20051203013941.GA2663@havoc.gtf.org>
Content-Disposition: inline

From: Jeff Garzik <jgarzik@pobox.com>

[x86, PCI] pass PCI domain number to PCI config read/write hooks

Don't hardcode zero, since modern x86 (with special ACPI sauce)
can support multiple "PCI segments", aka PCI domains.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/i386/pci/common.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/arch/i386/pci/common.c
+++ gregkh-2.6/arch/i386/pci/common.c
@@ -31,12 +31,14 @@ struct pci_raw_ops *raw_pci_ops;
 
 static int pci_read(struct pci_bus *bus, unsigned int devfn, int where, int size, u32 *value)
 {
-	return raw_pci_ops->read(0, bus->number, devfn, where, size, value);
+	return raw_pci_ops->read(pci_domain_nr(bus), bus->number,
+				 devfn, where, size, value);
 }
 
 static int pci_write(struct pci_bus *bus, unsigned int devfn, int where, int size, u32 value)
 {
-	return raw_pci_ops->write(0, bus->number, devfn, where, size, value);
+	return raw_pci_ops->write(pci_domain_nr(bus), bus->number,
+				  devfn, where, size, value);
 }
 
 struct pci_ops pci_root_ops = {
